packages/headless/src/features/commerce/context/cart/cart-slice.ts

```diff
@@ -0,0 +1,49 @@
+import {createReducer} from '@reduxjs/toolkit';
+import {CartState, getCartInitialState} from './cart-state';
+import {addCartItem, removeCartItem, setCart, updateCartItemQuantity} from './cart-actions';
+import {ProductParam} from '../../../../api/commerce/commerce-api-params';
+
+const generateId = (product: ProductParam) => [
```

Comment:
Related to my comment above about all properties being optional on product, this means we can generate empty string for an ID.